src/main/kotlin/com/github/hanseter/json/editor/types/FormattedIntegerModel.kt

```diff
@@ -0,0 +1,77 @@
+/*
+ * SWEETS - Software Engineering Tooling Suite
+ *
+ * Copyright (c) Siemens Mobility GmbH 2022, All Rights Reserved, Confidential.
+ */
+package com.github.hanseter.json.editor.types
+
+import com.github.hanseter.json.editor.controls.DoubleControl
+import com.github.hanseter.json.editor.extensions.EffectiveSchema
+import javafx.util.StringConverter
+import org.everit.json.schema.NumberSchema
+import java.text.DecimalFormat
+import java.text.DecimalFormatSymbols
+import java.text.ParsePosition
+import kotlin.math.pow
+import kotlin.math.roundToInt
+
+/**
+ *
+ * @author Henrik Frühling (henrik.fruehling@siemens.com)
+ */
+class FormattedIntegerModel(
+    schema: EffectiveSchema<NumberSchema>,
+    decimalFormatSymbols: DecimalFormatSymbols
+) : IntegerModel(schema) {
+
+    private val intFormat = IntFormat(schema.baseSchema.unprocessedProperties, decimalFormatSymbols)
+
+    override val previewString: PreviewString
+        get() = PreviewString.create(
+            intFormat.converter.toString(value),
+            intFormat.converter.toString(defaultValue),
+            rawValue
+        )
+
+    companion object {
+        const val INT_FORMAT = "int-format"
+        private const val INT_FORMAT_PATTERN = "pattern"
+        private const val INT_FORMAT_PRECISION = "precision"
+
+        class IntFormat(
```

Comment:
Why is this scoped under the companion object?

Reply:
better lift out this class to a separate file?